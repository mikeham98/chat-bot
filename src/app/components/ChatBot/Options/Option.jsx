import React from 'react';
import styles from '../../../../themes/components/messages.scss';
import classnames from "classnames";

export default class Option extends React.Component {
    constructor(props) {
        super(props);
        this.onClickOption = this.onClickOption.bind(this);
    }

    returnClassName() {
        const {selected, disabled} = this.props;
        const classNames = [styles.messageOption];
        if (selected) {
            classNames.push(styles.messageOptionSelected);
        } else if (disabled) {
            classNames.push(styles.messageOptionDisabled);
        }
        return classnames(classNames);
    }

    onClickOption(e) {
        const {onClick, disabled} = this.props;
        if (!disabled) {
            e.stopPropagation();
            onClick();
        }
    }

    render() {
        const {option} = this.props;
        return (
            <div className={this.returnClassName()} onClick={this.onClickOption}>
                {option}
            </div>
        );
    }
}