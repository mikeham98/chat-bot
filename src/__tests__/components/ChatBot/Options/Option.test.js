import React from 'react';
import {shallow} from 'enzyme';
import Option from '../../../../app/components/ChatBot/Options/Option';

describe('Option', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<Option {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            disabled: false,
            selected: false,
            option: 'Option name',
            onClick: jest.fn()
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a div with a className of messageOption', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a div with a className of messageOption and messageOptionSelected', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a div with a className of messageOption and messageOptionDisabled', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('prop', () => {
            describe('onClick', () => {
                it('should be called on onClick of div', () => {
                    const event = {
                        stopPropagation: jest.fn()
                    };
                    expect(event.stopPropagation).not.toHaveBeenCalled();
                    expect(props.onClick).not.toHaveBeenCalled();
                    wrapper().find('div').first().props().onClick(event);
                    expect(event.stopPropagation).toHaveBeenCalledTimes(1);
                    expect(props.onClick).toHaveBeenCalledTimes(1);
                });
                it('should not called on onClick of div because it is disabled', () => {
                    props.disabled = true;
                    const event = {
                        stopPropagation: jest.fn()
                    };
                    expect(event.stopPropagation).not.toHaveBeenCalled();
                    expect(props.onClick).not.toHaveBeenCalled();
                    wrapper().find('div').first().props().onClick(event);
                    expect(event.stopPropagation).not.toHaveBeenCalled();
                    expect(props.onClick).not.toHaveBeenCalled();
                });
            });
        });
    });
});