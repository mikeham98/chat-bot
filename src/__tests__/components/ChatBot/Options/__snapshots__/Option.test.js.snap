// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option render should return a div with a className of messageOption 1`] = `
<div
  className="messageOption"
  onClick={[Function]}
>
  Option name
</div>
`;

exports[`Option render should return a div with a className of messageOption and messageOptionDisabled 1`] = `
<div
  className="messageOption"
  onClick={[Function]}
>
  Option name
</div>
`;

exports[`Option render should return a div with a className of messageOption and messageOptionSelected 1`] = `
<div
  className="messageOption"
  onClick={[Function]}
>
  Option name
</div>
`;
